bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BFutureType.java

```diff
@@ -69,7 +69,12 @@ public int getTag() {
 
     @Override
     public boolean equals(Object obj) {
-        if (!super.equals(obj) || !(obj instanceof BTableType)) {
+
+        if (!(obj instanceof BFutureType)) {
+            return false;
+        }
+
+        if (!super.equals(obj)) {
```

Comment:
I think  if (!super.equals(obj))  check does not need. we can do same as BParameterizedType

Reply:
ex.        
```
        if (this == obj) {
            return true;
        }

        if (!(obj instanceof BFutureType)) {
            return false;
        }
```